yt_dlp/extractor/mx3.py

```diff
@@ -0,0 +1,160 @@
+import re
+
+from .common import InfoExtractor
+from ..utils.traversal import traverse_obj
+from ..utils import (
+    urlhandle_detect_ext,
+    url_or_none,
+    int_or_none,
+    unified_timestamp,
+)
+from ..networking import HEADRequest
+
+
+class Mx3BaseIE(InfoExtractor):
+    _MX3_DOMAIN = None
+
+    def _real_extract(self, url):
+        track_id = self._match_id(url)
+        webpage = self._download_webpage(url, track_id)
+        json = self._download_json(f'https://{self._MX3_DOMAIN}/t/{track_id}.json', track_id)
+
+        artists = []
+        if json.get('artist'):
+            artists.append(json['artist'])
+        performer = json.get('performer_name')
+        if performer and performer not in artists:
+            artists.append(performer)
+
+        genre = self._html_search_regex(r'<div\b[^>]+class="single-band-genre"[^>]*>([^<]+)</div>',
+                                        webpage, 'genre', fatal=False, flags=re.DOTALL)
+
+        formats = []
+
+        def add_format(fmt, fatal):
+            if fatal:
+                urlh = self._request_webpage(HEADRequest(fmt['url']), track_id, note='Fetching default media headers')
+            else:
+                urlh = self._request_webpage(HEADRequest(fmt['url']), track_id, fatal=False, expected_status=404,
+                                             note=f'Trying media headers for optional format {fmt["format_id"]}')
+            if urlh and urlh.status == 200:
+                fmt['ext'] = urlhandle_detect_ext(urlh)
+                fmt['filesize'] = int_or_none(urlh.headers.get('Content-Length'))
+                fmt['timestamp'] = unified_timestamp(urlh.headers.get('Last-Modified'))
+                formats.append(fmt)
+
+        track_url = f'https://{self._MX3_DOMAIN}/tracks/{track_id}'
+        add_format({
+            'url': f'{track_url}/player_asset',
+            'format_id': 'default',
+            'quality': 1,
+        }, fatal=True)
+        # the formats below don't always exist
+        add_format({
+            'url': f'{track_url}/player_asset?quality=hd',
+            'format_id': 'hd',
+            'quality': 10,
+        }, fatal=False)
+        add_format({
+            'url': f'{track_url}/download',
+            'format_id': 'download',
+            'quality': 11,
+        }, fatal=False)
```

Comment:
Hm. I can see your point. But for me there is simply no difference in quality between a high-bitrate MP3 and WAV. For video I would list the formats and pick one manually, but for music I use the "Open With" browser extension (non-interactive) and check the download folder later. I'm going to use `-f"best[ext!=wav][ext!=flac][filesize<50M]" -x` now, so it will work for me if you add the format.

Reply:
If you only want the format selection for this site, just use `-f hd/default`. This will download `hd` if available and otherwise fallback to `default`.